flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/topic/range/RangeGenerator.java

```diff
@@ -51,7 +51,18 @@ public interface RangeGenerator extends Serializable {
      */
     List<TopicRange> range(TopicMetadata metadata, int parallelism);
 
-    /** Initialize some extra resource when bootstrap the source. */
+    /**
+     * Defines the default behavior for Key_Shared subscription in Flink. See {@link KeySharedMode}
+     * for the detailed usage of the key share mode.
+     *
+     * @param metadata The metadata of the topic.
+     * @param parallelism The reader size for this topic.
+     */
+    default KeySharedMode keyShareMode(TopicMetadata metadata, int parallelism) {
```

Comment:
It seems no implementation use the parameters.

Reply:
This is kept only for end-user if they needs.